### PR TITLE
Lesson5-7: Character Device 

### PR DESCRIPTION
**Pull request by daaaanil81**
Implement a character device driver for text messaging between users.

1. Implement a character device driver with the following requirements:
- 1kB buffer size;
- buffer increase as a module parameter ;
- should be available for all users;
- works only ASCII strings.

2.  Add the interfaces into the driver:
- sysfs for buffer clean up;
- procfs for displaying used buffer volume and buffer size

Bash script will be later. 
@galkinletter Check please my task.

### CODE REVIEW

File: 02-KernelModules_CharacterDevice/moduleCharDev.c [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+static ssize_t proc_read(struct file*, char __user*, size_t len, loff_t*);
+struct class_attribute sysfs = __ATTR(SYSFS_NAME, 0664, &sysfs_show, &sysfs_store);
+module_param(all_memory, int, 0664);
+static int create_buffer(void)
+{
+    all_memory += BUFFER_SIZE;
+    dev_buffer = kmalloc(all_memory, GFP_KERNEL);
+    if (NULL == dev_buffer)
+        return -ENOMEM;
+    memset(dev_buffer, 0, all_memory);
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/moduleCharDev.c]: Use kzalloc() to allocate memory. The memory is set to zero.

File: 02-KernelModules_CharacterDevice/moduleCharDev.c [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    return 0;
+}
+
+
+static void cleanup_buffer(void)
+{
+    if (dev_buffer) {
+        memset(dev_buffer, 0, all_memory);
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/moduleCharDev.c]: Use kzfree() to wipe memory

File: 02-KernelModules_CharacterDevice/moduleCharDev.c [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+{
+    int len = strlen(buf) - 1;
+    unsigned int time;
+    int i;
+    printk(KERN_INFO "Check_ascii: %s and size: %d\n", buf, len);
+    for (i = 0; i < len; i++)
+    {
+        time = buf[i];
+        printk(KERN_INFO "%d", time);
+        if (time < 33 || time > 126) {
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/moduleCharDev.c]: Why space is disabled here? ASCII 32=SPACE

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
Plz make those minor fixes.

**Review by galkinletter — CHANGES_REQUESTED** [review]
Please fix

**Comment by daaaanil81** [comment]
@galkinletter @oleg-h2o Check please my task.

**Comment by galkinletter** [comment]
Please Bash script to do it  also. .

**Comment by galkinletter** [comment]
Driver is OK. Bugs is fixed

**Pull request merged**